apps/sms/views/conversation/js/information.js

```diff
@@ -357,6 +366,48 @@ var VIEWS = {
       }).bind(this);
     },
 
+    renderLateArrivalMsg: function report_renderLateArrivalMsg(message) {
+      var latenessBlock = this.panel.querySelector('.lateness-block');
+      var lateness = message.timestamp - message.sentTimestamp;
+      var showLatenessNotice = message.timestamp &&
+        message.sentTimestamp &&
+        lateness >= LATENESS_DISPLAY_LOWER_LIMIT;
+
+      if (showLatenessNotice) {
+        var units = [
+          { unit: 'months', unitVal: MONTH },
+          { unit: 'days', unitVal: DAY },
+          { unit: 'hours', unitVal: HOUR },
+          { unit: 'minutes', unitVal: MINUTE },
+          { unit: 'seconds', unitVal: SECOND }
+        ];
+
+        units.forEach(({unit, unitVal}) => {
+          var output;
+          var unitElement = this.panel.querySelector('.lateness-' + unit);
+
+          if (lateness >= unitVal) {
+            // Compute integer value for the current unit output 
+            // and get the rest for the next smaller unit
+            output = Math.floor(lateness / unitVal);
+            lateness = lateness % unitVal;
+            
+            navigator.mozL10n.setAttributes(
+              unitElement,
+              'message-lateness-' + unit, 
+              { [unit]: output }
+            );
+            unitElement.classList.remove('hide');
+          } else {
+            unitElement.classList.add('hide');
+          }
+        });
+        latenessBlock.classList.remove('hide');
+      } else {
+        latenessBlock.classList.add('hide');
+      }
+    },
+
```

Comment:
sorry that I see this this late as well... but I see no unit test for the changes in this file ? Can you write some ?